### DISCUSSION TITLE
How to use xautoclaim gracefully?

### DISCUSSION THREAD

**Discussion by nyongja** [redis/go-redis#2660] [Q&A]
Hello, I would like to apply redis streams.
I'm going to make one consumer into one goroutine.
One consumer repeatedly reads the message to the `xreadgroup` through the for statement.
At this time, I would like to use the `xautoclaim` command to process messages that are pending for a certain period of time.

The percentage of messages in the pending state will be very low, but simply calling xreadgroup and xautoclaim alternately seems inefficient as shown below..
```
func consume() {
    for {
        - redisXReadGroup 
        - redis.RedisXAutoClaim
     }
}
```

Should there be other objects running xautoclaim periodically?
Or is there any way?

**Comment by elpablete** [comment]
Why would you call that inefficient?
